p4-16/spec/P4-16-spec.mdk

```diff
@@ -3731,6 +3733,10 @@ the compiler will add implicit casts as follows:
 - `x << 13` becomes `0`; overflow warning
 - `x | 0xFFF` becomes `x | (bit<8>)0xFFF`; overflow warning
 - `x + E.a` becomes `x + (bit<8>)E.a`
+- `16w11 << E.a` becomes `16w11 << (bit<8>)E.a`
+- `E.a[3:0]` becomes `((bit<8>)E.a)[3:0]`
```

Comment:
We should remove this line, and explain it in a separate PR addressing slices, l-values, and serializable enums.

Reply:
I have created this PR regarding this issue: https://github.com/p4lang/p4-spec/issues/992.  Please take a look at it and add corrections or clarifications to my initial attempt at describing it.